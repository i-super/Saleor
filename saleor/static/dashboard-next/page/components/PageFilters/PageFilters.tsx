import TextField from "material-ui/TextField";
import * as React from "react";

import FilterCard from "../../../components/FilterCard";
import FormSpacer from "../../../components/FormSpacer";
import i18n from "../../../i18n";

interface PageFiltersState {
  title: string;
  url: string;
  debounceTimeout: any;
}

interface PageFiltersProps {
  formState?: {
    title: string;
    url: string;
  };
  handleClear();
  handleSubmit(formState: any);
}

export class PageFilters extends React.Component<
  PageFiltersProps,
  PageFiltersState
> {
  static defaultState = {
    title: "",
    url: ""
  };

  constructor(props) {
    super(props);
    this.state = { ...PageFilters.defaultState, ...props.formState };
  }

  render() {
    const { handleClear, handleSubmit } = this.props;
    const handleInputChange = event => {
      const { name, value } = event.target;
      const debounceTimeout = setTimeout(() => {
        handleSubmit({
          formData: { [name]: value }
        });
      }, 500);
      clearTimeout(this.state.debounceTimeout);
      this.setState({
        [name]: value,
        debounceTimeout
      });
    };

    const handleClearButtonClick = () => {
      this.setState(PageFilters.defaultState);
      handleClear();
    };
    return (
      <FilterCard handleClear={handleClearButtonClick}>
        <TextField
          fullWidth
          name="title"
          label={i18n.t("Title", {
            context: "Product filter field label"
          })}
          onChange={handleInputChange}
          value={this.state.title || ""}
        />
        <FormSpacer />
        <TextField
          fullWidth
          name="url"
          label={i18n.t("Url", {
            context: "Product filter field label"
          })}
          onChange={handleInputChange}
          value={this.state.url || ""}
        />
      </FilterCard>
    );
  }
}

export default PageFilters;
