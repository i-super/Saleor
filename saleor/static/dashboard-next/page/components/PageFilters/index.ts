export { default } from "./PageFilters";
export * from "./PageFilters";
