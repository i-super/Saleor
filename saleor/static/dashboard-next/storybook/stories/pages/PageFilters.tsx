import { storiesOf } from "@storybook/react";
import * as React from "react";

import PageFilters from "../../../page/components/PageFilters";

storiesOf("Pages / PageFilters", module)
  .add("with initial data", () => (
    <PageFilters
      handleClear={() => {}}
      handleSubmit={() => {}}
      formState={{ title: "Initial title", url: "initial-url" }}
    />
  ))
  .add("without initial data", () => (
    <PageFilters handleClear={() => {}} handleSubmit={() => {}} />
  ));
