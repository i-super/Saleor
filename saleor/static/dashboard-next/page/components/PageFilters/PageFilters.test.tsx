import * as React from "react";
import * as renderer from "react-test-renderer";

import PageFilters from "./";

const pageFilters = {
  title: "title",
  url: "url"
};

describe("<PageFilters />", () => {
  it("renders without initial state", () => {
    const component = renderer.create(
      <PageFilters handleClear={jest.fn()} handleSubmit={jest.fn()} />
    );
    expect(component).toMatchSnapshot();
  });
  it("renders with initial state", () => {
    const component = renderer.create(
      <PageFilters
        formState={pageFilters}
        handleClear={jest.fn()}
        handleSubmit={jest.fn()}
      />
    );
    expect(component).toMatchSnapshot();
  });
});
