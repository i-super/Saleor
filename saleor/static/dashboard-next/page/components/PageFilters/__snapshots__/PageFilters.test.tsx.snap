// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PageFilters /> renders with initial state 1`] = `
<div
  className="MuiPaper-root-1 MuiPaper-elevation2-5 MuiPaper-rounded-2"
>
  <form>
    <div
      className="MuiCardHeader-root-28"
    >
      <div
        className="MuiCardHeader-content-31"
      >
        <span
          className="MuiTypography-root-34 MuiTypography-headline-39 MuiCardHeader-title-32"
        >
          Filters
        </span>
      </div>
      <div
        className="MuiCardHeader-action-30"
      >
        <button
          className="MuiButtonBase-root-64 MuiIconButton-root-58"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex="0"
          type="button"
        >
          <span
            className="MuiIconButton-label-63"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-67"
              color={undefined}
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M17.65 6.35C16.2 4.9 14.21 4 12 4c-4.42 0-7.99 3.58-7.99 8s3.57 8 7.99 8c3.73 0 6.84-2.55 7.73-6h-2.08c-.82 2.33-3.04 4-5.65 4-3.31 0-6-2.69-6-6s2.69-6 6-6c1.66 0 3.14.69 4.22 1.78L13 11h7V4l-2.35 2.35z"
              />
            </svg>
          </span>
          <span
            className="MuiTouchRipple-root-73"
          />
        </button>
      </div>
    </div>
    <div
      className="MuiCardContent-root-80"
    >
      <div
        aria-describedby={undefined}
        className="MuiFormControl-root-81 MuiFormControl-fullWidth-84"
        onBlur={[Function]}
        onFocus={[Function]}
      >
        <label
          className="MuiFormLabel-root-91 MuiInputLabel-root-85 MuiInputLabel-formControl-86 MuiInputLabel-animated-89 MuiInputLabel-shrink-88"
          data-shrink={true}
          htmlFor={undefined}
        >
          Title
        </label>
        <div
          className="MuiInput-root-97 MuiInput-fullWidth-104 MuiInput-formControl-98 MuiInput-underline-101"
        >
          <input
            aria-invalid={false}
            aria-required={false}
            autoComplete={undefined}
            autoFocus={undefined}
            className="MuiInput-input-105"
            defaultValue={undefined}
            disabled={false}
            id={undefined}
            name="title"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={undefined}
            onKeyUp={undefined}
            placeholder={undefined}
            readOnly={undefined}
            required={undefined}
            rows={undefined}
            type="text"
            value="title"
          />
        </div>
      </div>
      <div
        className="Component-spacer-111"
      />
      <div
        aria-describedby={undefined}
        className="MuiFormControl-root-81 MuiFormControl-fullWidth-84"
        onBlur={[Function]}
        onFocus={[Function]}
      >
        <label
          className="MuiFormLabel-root-91 MuiInputLabel-root-85 MuiInputLabel-formControl-86 MuiInputLabel-animated-89 MuiInputLabel-shrink-88"
          data-shrink={true}
          htmlFor={undefined}
        >
          Url
        </label>
        <div
          className="MuiInput-root-97 MuiInput-fullWidth-104 MuiInput-formControl-98 MuiInput-underline-101"
        >
          <input
            aria-invalid={false}
            aria-required={false}
            autoComplete={undefined}
            autoFocus={undefined}
            className="MuiInput-input-105"
            defaultValue={undefined}
            disabled={false}
            id={undefined}
            name="url"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={undefined}
            onKeyUp={undefined}
            placeholder={undefined}
            readOnly={undefined}
            required={undefined}
            rows={undefined}
            type="text"
            value="url"
          />
        </div>
      </div>
    </div>
  </form>
</div>
`;

exports[`<PageFilters /> renders without initial state 1`] = `
<div
  className="MuiPaper-root-1 MuiPaper-elevation2-5 MuiPaper-rounded-2"
>
  <form>
    <div
      className="MuiCardHeader-root-28"
    >
      <div
        className="MuiCardHeader-content-31"
      >
        <span
          className="MuiTypography-root-34 MuiTypography-headline-39 MuiCardHeader-title-32"
        >
          Filters
        </span>
      </div>
      <div
        className="MuiCardHeader-action-30"
      >
        <button
          className="MuiButtonBase-root-64 MuiIconButton-root-58"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex="0"
          type="button"
        >
          <span
            className="MuiIconButton-label-63"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-67"
              color={undefined}
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M17.65 6.35C16.2 4.9 14.21 4 12 4c-4.42 0-7.99 3.58-7.99 8s3.57 8 7.99 8c3.73 0 6.84-2.55 7.73-6h-2.08c-.82 2.33-3.04 4-5.65 4-3.31 0-6-2.69-6-6s2.69-6 6-6c1.66 0 3.14.69 4.22 1.78L13 11h7V4l-2.35 2.35z"
              />
            </svg>
          </span>
          <span
            className="MuiTouchRipple-root-73"
          />
        </button>
      </div>
    </div>
    <div
      className="MuiCardContent-root-80"
    >
      <div
        aria-describedby={undefined}
        className="MuiFormControl-root-81 MuiFormControl-fullWidth-84"
        onBlur={[Function]}
        onFocus={[Function]}
      >
        <label
          className="MuiFormLabel-root-91 MuiInputLabel-root-85 MuiInputLabel-formControl-86 MuiInputLabel-animated-89"
          data-shrink={false}
          htmlFor={undefined}
        >
          Title
        </label>
        <div
          className="MuiInput-root-97 MuiInput-fullWidth-104 MuiInput-formControl-98 MuiInput-underline-101"
        >
          <input
            aria-invalid={false}
            aria-required={false}
            autoComplete={undefined}
            autoFocus={undefined}
            className="MuiInput-input-105"
            defaultValue={undefined}
            disabled={false}
            id={undefined}
            name="title"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={undefined}
            onKeyUp={undefined}
            placeholder={undefined}
            readOnly={undefined}
            required={undefined}
            rows={undefined}
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        className="Component-spacer-111"
      />
      <div
        aria-describedby={undefined}
        className="MuiFormControl-root-81 MuiFormControl-fullWidth-84"
        onBlur={[Function]}
        onFocus={[Function]}
      >
        <label
          className="MuiFormLabel-root-91 MuiInputLabel-root-85 MuiInputLabel-formControl-86 MuiInputLabel-animated-89"
          data-shrink={false}
          htmlFor={undefined}
        >
          Url
        </label>
        <div
          className="MuiInput-root-97 MuiInput-fullWidth-104 MuiInput-formControl-98 MuiInput-underline-101"
        >
          <input
            aria-invalid={false}
            aria-required={false}
            autoComplete={undefined}
            autoFocus={undefined}
            className="MuiInput-input-105"
            defaultValue={undefined}
            disabled={false}
            id={undefined}
            name="url"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={undefined}
            onKeyUp={undefined}
            placeholder={undefined}
            readOnly={undefined}
            required={undefined}
            rows={undefined}
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
  </form>
</div>
`;
